quartz/components/scripts/search.inline.ts

```diff
@@ -71,20 +71,44 @@ function highlight(searchTerm: string, text: string, trim?: boolean) {
   }`
 }
 
+const p = new DOMParser()
 const encoder = (str: string) => str.toLowerCase().split(/([^a-z]|[^\x00-\x7F])/)
 let prevShortcutHandler: ((e: HTMLElementEventMap["keydown"]) => void) | undefined = undefined
-document.addEventListener("nav", async (e: unknown) => {
-  const currentSlug = (e as CustomEventMap["nav"]).detail.url
+
+let fetchContentCache: Record<string, Element[] | undefined> = {}
```

Comment:
```suggestion
const fetchContentCache: Map<FullSlug, Element[]> = new Map()
```

prefer `Map` for mutable